drivers/dai/intel/dmic/dmic.c

```diff
@@ -895,13 +895,8 @@ static int dai_dmic_initialize_device(const struct device *dev)
 		dai_dmic_irq_handler,
 		DEVICE_DT_INST_GET(0),
 		0);
-	if (pm_device_on_power_domain(dev)) {
-		pm_device_init_off(dev);
-	} else {
-		pm_device_init_suspended(dev);
-	}
 
-	return pm_device_runtime_enable(dev);
+	return pm_device_driver_init(dev, dmic_pm_action);
```

Comment:
Not an expert with the pm code, but this does look very clean and matches with other drivers' use of Zephyr runtime-pm, so looks good for me.